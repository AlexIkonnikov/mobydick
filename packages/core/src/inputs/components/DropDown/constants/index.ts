import rem from '../../../../styles/utils/rem';

export const BORDER_BUTTON_WIDTH = rem(1);

export const DROP_DOWN_POPUP_ID = 'DropDownPopup';

export const DEFAULT_DROP_DOWN_HEIGHT = rem(48);
export const DEFAULT_DROP_DOWN_WIDTH = rem(200);
export const DEFAULT_DROPDOWN_MARGIN_FROM_BUTTON = rem(4);

export const LIST_MAX_HEIGHT = rem(244);
export const EMPTY_LIST_HEIGHT = rem(74);
