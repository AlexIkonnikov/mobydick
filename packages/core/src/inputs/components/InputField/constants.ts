enum Constants {
  testID = 'TextInput',
}

export default Constants;
