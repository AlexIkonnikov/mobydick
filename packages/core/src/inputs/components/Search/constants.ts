export const strings = {
  search: 'Поиск',
};
