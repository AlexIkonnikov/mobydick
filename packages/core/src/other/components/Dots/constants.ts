import rem from '../../../styles/utils/rem';

export const SPAN_SIZE = 3;

export const SIZE_LARGE = rem(8);
export const SIZE_MEDIUM = rem(6);
export const SIZE_SMALL = rem(4);

export const MARGIN_DOT = rem(5) * 2;

export const WIDTH_SMALL =
  (SIZE_LARGE + MARGIN_DOT) * 3 +
  (SIZE_MEDIUM + MARGIN_DOT) +
  (SIZE_SMALL + MARGIN_DOT);

export const WIDTH_MEDIUM = WIDTH_SMALL + (SIZE_MEDIUM + MARGIN_DOT);

export const WIDTH_LARGE = WIDTH_MEDIUM + (SIZE_SMALL + MARGIN_DOT);
