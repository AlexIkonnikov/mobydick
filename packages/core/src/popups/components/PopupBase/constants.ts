enum Constants {
  testID = 'Pressable',
}

export default Constants;
