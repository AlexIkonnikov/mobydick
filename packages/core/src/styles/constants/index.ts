import HIT_SLOP from './HIT_SLOP';

export {HIT_SLOP};
