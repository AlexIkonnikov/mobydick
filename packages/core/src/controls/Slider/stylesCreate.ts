import {I18nManager, StyleSheet} from 'react-native';

import {IThemeContext} from '../../styles';

const stylesCreate = ({colors, spaces}: IThemeContext) => {
  return StyleSheet.create({
    controlsContainer: {
      flexDirection: 'row',
      justifyContent: I18nManager.isRTL ? 'flex-end' : 'flex-start',
      alignItems: 'center',
      minHeight: spaces.Space32,
    },
    rail: {
      flex: 1,
      height: spaces.Space4,
      borderRadius: spaces.Space2,
      backgroundColor: colors.IconMuted,
    },
    selectedRailContainer: {
      position: 'absolute',
    },
    selectedRail: {
      height: spaces.Space4,
      backgroundColor: colors.IconBase,
      borderRadius: spaces.Space2,
    },
    highThumbContainer: {
      position: 'absolute',
    },
    railsContainer: {
      ...StyleSheet.absoluteFillObject,
      flexDirection: 'row',
      alignItems: 'center',
    },
    touchableArea: {
      ...StyleSheet.absoluteFillObject,
    },
  });
};

export default stylesCreate;
