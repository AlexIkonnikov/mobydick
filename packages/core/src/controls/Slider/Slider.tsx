import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
import {
  Animated,
  GestureResponderEvent,
  Insets,
  PanResponder,
  PanResponderGestureState,
  ViewProps,
  LayoutChangeEvent,
} from 'react-native';

import View from '../../basic/components/View/View';
import useStyles from '../../styles/theme/hooks/useStyles';
import {LABELS} from '../../other/constants';
import {HIT_SLOP} from '../../styles';

import {clamp, getValueForPosition, isLowCloser} from './helpers';
import stylesCreate from './stylesCreate';
import {useLowHigh, useSelectedRail, useWidthLayout} from './hooks';
import HighThumb from './components/HighThumb';
import Rail from './components/Rail';
import LowThumb from './components/LowThumb';

const trueFunc = () => true;
const falseFunc = () => false;

export interface ISliderProps extends ViewProps {
  min: number;
  max: number;
  step: number;
  minRange?: number;
  low?: number;
  high?: number;
  disableRange?: boolean;
  disabled?: boolean;
  onValueChanged?: (low: number, high: number, byUser: boolean) => void;
  hitSlop?: Insets;
}
const Slider: React.FC<ISliderProps> = ({
  min,
  max,
  minRange = 0,
  step,
  low: lowProp,
  high: highProp,
  disableRange = false,
  disabled = false,
  onValueChanged,
  hitSlop = HIT_SLOP.large,
  ...restProps
}) => {
  const {inPropsRef, inPropsRefPrev, setLow, setHigh} = useLowHigh(
    lowProp,
    disableRange ? max : highProp,
    min,
    max,
    step,
  );
  const [styles] = useStyles(stylesCreate);

  const lowThumbXRef = useRef(new Animated.Value(0));
  const highThumbXRef = useRef(new Animated.Value(0));
  const pointerX = useRef(new Animated.Value(0)).current;
  const {current: lowThumbX} = lowThumbXRef;
  const {current: highThumbX} = highThumbXRef;

  const gestureStateRef = useRef({isLow: true, lastValue: 0, lastPosition: 0});

  const containerWidthRef = useRef(0);
  const [thumbWidth, setThumbWidth] = useState(0);

  const [selectedRailStyle, updateSelectedRail] = useSelectedRail(
    inPropsRef,
    containerWidthRef,
    thumbWidth,
    disableRange,
  );

  const updateThumbs = useCallback(() => {
    const {current: containerWidth} = containerWidthRef;

    const {low, high} = inPropsRef.current;
    if (!disableRange) {
      const {current: highThumbX} = highThumbXRef;
      const highPosition =
        ((high - min) / (max - min)) * (containerWidth - thumbWidth);
      highThumbX.setValue(highPosition);
    }
    const {current: lowThumbX} = lowThumbXRef;
    const lowPosition =
      ((low - min) / (max - min)) * (containerWidth - thumbWidth);
    lowThumbX.setValue(lowPosition);
    updateSelectedRail();
    onValueChanged?.(low, high, false);
  }, [disableRange, inPropsRef, max, min, thumbWidth, updateSelectedRail]);

  useEffect(() => {
    const {lowPrev, highPrev} = inPropsRefPrev;
    if (
      (lowProp !== undefined && lowProp !== lowPrev) ||
      (highProp !== undefined && highProp !== highPrev)
    ) {
      updateThumbs();
    }
  }, [highProp, inPropsRefPrev.lowPrev, inPropsRefPrev.highPrev, lowProp]);

  useEffect(() => {
    updateThumbs();
  }, [updateThumbs]);

  const handleContainerLayout = useWidthLayout(containerWidthRef, updateThumbs);
  const handleThumbLayout = useCallback(
    (event: LayoutChangeEvent) => {
      const {
        layout: {width},
      } = event.nativeEvent;
      if (thumbWidth !== width) {
        setThumbWidth(width);
      }
    },
    [thumbWidth],
  );

  function getLow(downX: number, lowPosition: number, highPosition: number) {
    return disableRange || isLowCloser(downX, lowPosition, highPosition);
  }

  const handlePositionChange = (
    positionInView: number,
    isLow: boolean,
    containerWidth: number,
  ) => {
    const {low, high, min, max, step} = inPropsRef.current;
    const minValue = isLow ? min : low + minRange;
    const maxValue = isLow ? high - minRange : max;
    const value = clamp(
      getValueForPosition(
        positionInView,
        containerWidth,
        thumbWidth,
        min,
        max,
        step,
      ),
      minValue,
      maxValue,
    );
    if (gestureStateRef.current.lastValue === value) {
      return;
    }
    const availableSpace = containerWidth - thumbWidth;
    const absolutePosition = ((value - min) / (max - min)) * availableSpace;
    gestureStateRef.current.lastValue = value;
    gestureStateRef.current.lastPosition = absolutePosition + thumbWidth / 2;
    (isLow ? lowThumbX : highThumbX).setValue(absolutePosition);
    onValueChanged?.(isLow ? value : low, isLow ? high : value, true);
    (isLow ? setLow : setHigh)(value);
    updateSelectedRail();
  };

  const {panHandlers} = useMemo(
    () =>
      PanResponder.create({
        onStartShouldSetPanResponderCapture: falseFunc,
        onMoveShouldSetPanResponderCapture: falseFunc,
        onPanResponderTerminationRequest: falseFunc,
        onPanResponderTerminate: trueFunc,
        onShouldBlockNativeResponder: trueFunc,
        onMoveShouldSetPanResponder: (
          _evt: GestureResponderEvent,
          gestureState: PanResponderGestureState,
        ) => Math.abs(gestureState.dx) > 2 * Math.abs(gestureState.dy),

        onPanResponderGrant: ({nativeEvent}, gestureState) => {
          if (disabled) {
            return;
          }
          const {numberActiveTouches} = gestureState;
          if (numberActiveTouches > 1) {
            return;
          }

          const {locationX: downX, pageX} = nativeEvent;
          const containerX = pageX - downX;

          const {low, high, min, max} = inPropsRef.current;
          const containerWidth = containerWidthRef.current;

          const lowPosition =
            thumbWidth / 2 +
            ((low - min) / (max - min)) * (containerWidth - thumbWidth);
          const highPosition =
            thumbWidth / 2 +
            ((high - min) / (max - min)) * (containerWidth - thumbWidth);
          const isLow = getLow(downX, lowPosition, highPosition);

          gestureStateRef.current.isLow = isLow;

          handlePositionChange(downX, isLow, containerWidth);
          pointerX.removeAllListeners();
          pointerX.addListener(({value: pointerPosition}) => {
            const positionInView = pointerPosition - containerX;
            handlePositionChange(positionInView, isLow, containerWidth);
          });
        },

        onPanResponderMove: disabled
          ? undefined
          : Animated.event([null, {moveX: pointerX}], {useNativeDriver: false}),
      }),
    [
      pointerX,
      inPropsRef,
      thumbWidth,
      disableRange,
      disabled,
      onValueChanged,
      setLow,
      setHigh,
      updateSelectedRail,
    ],
  );

  return (
    <View {...restProps}>
      <View onLayout={handleContainerLayout} style={styles.controlsContainer}>
        <Rail selectedRailStyle={selectedRailStyle} thumbWidth={thumbWidth} />
        <LowThumb handleThumbLayout={handleThumbLayout} lowThumbX={lowThumbX} />
        {!disableRange && <HighThumb highThumbX={highThumbX} />}
        <View
          {...panHandlers}
          hitSlop={hitSlop}
          style={styles.touchableArea}
          accessibilityLabel={LABELS.slider}
          collapsable={false}
        />
      </View>
    </View>
  );
};

export default Slider;
