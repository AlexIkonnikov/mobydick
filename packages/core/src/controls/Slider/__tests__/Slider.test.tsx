import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';

import Slider from '../Slider';
import {LABELS} from '../../../other/constants';

const eventMock = {};
const trueFunc = () => true;
const falseFunc = () => false;

describe('Slider', () => {
  it('should renders correctly by default', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={0} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    act(() => panHandler.props.onMoveShouldSetPanResponder({}, {dx: -10}));
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with more props', function () {
    const {getByLabelText, toJSON} = render(
      <Slider
        min={0}
        max={100}
        step={1}
        minRange={10}
        low={20}
        high={60}
        disableRange={true}
      />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    act(() => panHandler.props.onMoveShouldSetPanResponder({}, {dx: -10}));
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly panHandler', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={0} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);

    act(() => {
      panHandler.props.onStartShouldSetPanResponderCapture(falseFunc);
      panHandler.props.onMoveShouldSetPanResponderCapture(falseFunc);
      panHandler.props.onPanResponderTerminationRequest(falseFunc);
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 33,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderMove(eventMock, {moveX: 20});
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly disabled', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={0} max={100} step={1} disableRange={true} disabled={true} />,
    );
    const panHandler = getByLabelText(LABELS.slider);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 33,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly updateThumbs', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={20} max={100} step={1} disableRange={true} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    const layout = getByLabelText(LABELS.sliderLayoutLowThumb);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 33,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderMove(eventMock, {moveX: 40});
    });
    expect(toJSON()).toMatchSnapshot();
    act(() => {
      fireEvent(layout, 'layout', {
        nativeEvent: {layout: {width: 100}},
      });
    });
  });
});
