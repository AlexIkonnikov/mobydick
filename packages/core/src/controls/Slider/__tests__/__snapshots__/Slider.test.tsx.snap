// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should renders correctly by default 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 0,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      onLayout={[Function]}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 18,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 24,
            "borderWidth": 3,
            "height": 48,
            "width": 48,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      onMoveShouldSetPanResponder={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderTerminate={[Function]}
      onShouldBlockNativeResponder={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly disabled 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 0,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      onLayout={[Function]}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 18,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      onMoveShouldSetPanResponder={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderTerminate={[Function]}
      onShouldBlockNativeResponder={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly panHandler 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 0,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": NaN,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      onLayout={[Function]}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 18,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": NaN,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 24,
            "borderWidth": 3,
            "height": 48,
            "width": 48,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      onMoveShouldSetPanResponder={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderTerminate={[Function]}
      onShouldBlockNativeResponder={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly updateThumbs 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 0,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": NaN,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      onLayout={[Function]}
      style={
        {
          "transform": [
            {
              "translateX": NaN,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 18,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      onMoveShouldSetPanResponder={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderTerminate={[Function]}
      onShouldBlockNativeResponder={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly with more props 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 0,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      onLayout={[Function]}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 18,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      onMoveShouldSetPanResponder={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderTerminate={[Function]}
      onShouldBlockNativeResponder={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
